src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -8866,12 +8866,17 @@ int RGWRados::cls_bucket_list_ordered(const DoutPrefixProvider *dpp,
       r = 0;
     }
 
+    // dirent might be moved in the next if statement
+    // This key needs to be copied so it can be logged later
+    string dirent_key = dirent.key;
+
     // at this point either r >= 0 or r == -ENOENT
     if (r >= 0) { // i.e., if r != -ENOENT
       ldpp_dout(dpp, 10) << __func__ << ": got " <<
 	dirent.key << dendl;
 
       auto [it, inserted] = m.insert_or_assign(name, std::move(dirent));
```

Comment:
why is "move" needed here at the first place?
don't we perform a copy when we insert or asign? if we do, can we pass `cosnt&` to the function instead of doing the move